topics/data-structure/index.md

```diff
@@ -0,0 +1,8 @@
+---
+aliases: data-structures
+display_name: Data structure
+short_description: Data structure is way of organizing and storing data.
+topic: data-structure
```

Comment:
`data-structures` is listed in https://github.com/github/explore/blob/master/topics-todo.md as one of the topics we want to curate, so thank you! Would you mind renaming this directory to data-structures/ and make `data-structure` an alias of `data-structures`? Then this line can be `topic: data-structures` and you can check off `data-structures` in the topics-todo.md file in your branch.